### PR TITLE
Add test for eta and mitigate float error.

### PR DESCRIPTION
**Pull request by trivialfis**

### CODE REVIEW

**Comment by trivialfis** [comment]
@hcho3 I think I got the CUDA test working again.

File: tests/cpp/tree/test_tree_stat.cc [hunk 1]
```diff
@@ -56,4 +56,62 @@ TEST_F(UpdaterTreeStatTest, Exact) {
...
+      {
+        tree_1.param.num_feature = kCols;
+        up_1->Update(&gpairs_, p_dmat_.get(), {&tree_1});
+      }
+      tree_0.WalkTree([&](bst_node_t nidx) {
+        if (tree_0[nidx].IsLeaf()) {
+          EXPECT_NEAR(tree_1[nidx].LeafValue() * eta, tree_0[nidx].LeafValue(), kRtEps);
+        }
+        return true;
+      });
```
Comment (hcho3) [inline tests/cpp/tree/test_tree_stat.cc:103]: How are these two trees identical when `eta` is not the same ?

Comment (trivialfis) [reply tests/cpp/tree/test_tree_stat.cc:103]: Notice the multiplication inside the compassion.

Comment (hcho3) [reply tests/cpp/tree/test_tree_stat.cc:103]: Doh! I didn't see that the first time.

**Review by hcho3 — APPROVED** [review]

**Review by RAMitchell — APPROVED** [review]

**Pull request merged**